From f85f2de174779aaf02cf2a858b2ba28c4cb6f843 Mon Sep 17 00:00:00 2001
From: Peter Korsgaard <jacmet@sunsite.dk>
Date: Sun, 25 Jan 2009 20:19:01 +0000
Subject: Makefile: globally define variables used at different places

$(comma) and $(space) are used in packages and the toolchain.

Signed-off-by: Markus Heidelberg <markus.heidelberg@web.de>
---
diff --git a/Makefile b/Makefile
index 6b12c76..a55d9bb 100644
--- a/Makefile
+++ b/Makefile
@@ -40,6 +40,11 @@ else
 BR2_LOCAL:=$(TOPDIR)/local
 endif
 
+# Variables for use in Make constructs
+comma:=,
+empty:=
+space:=$(empty) $(empty)
+
 # $(shell find . -name *_defconfig |sed 's/.*\///')
 # Pull in the user's configuration file
 ifeq ($(filter $(noconfig_targets),$(MAKECMDGOALS)),)
diff --git a/package/directfb/directfb.mk b/package/directfb/directfb.mk
index cbf15bc..1aa38c7 100644
--- a/package/directfb/directfb.mk
+++ b/package/directfb/directfb.mk
@@ -52,9 +52,6 @@ endif
 ifeq ($(DIRECTFB_GFX),)
 DIRECTFB_GFX:=none
 else
-comma:=,
-empty:=
-space:=$(empty) $(empty)
 DIRECTFB_GFX:=$(subst $(space),$(comma),$(strip $(DIRECTFB_GFX)))
 endif
 
@@ -75,9 +72,6 @@ endif
 ifeq ($(DIRECTFB_INPUT),)
 DIRECTFB_INPUT:=none
 else
-comma:=,
-empty:=
-space:=$(empty) $(empty)
 DIRECTFB_INPUT:=$(subst $(space),$(comma),$(strip $(DIRECTFB_INPUT)))
 endif
 
diff --git a/toolchain/gcc/gcc-uclibc-3.x.mk b/toolchain/gcc/gcc-uclibc-3.x.mk
index fc59b6b..368ac8e 100644
--- a/toolchain/gcc/gcc-uclibc-3.x.mk
+++ b/toolchain/gcc/gcc-uclibc-3.x.mk
@@ -233,7 +233,6 @@ gcc_initial-dirclean:
 # affect gcc-target. However, I haven't tested gcc-target yet so no
 # guarantees. mjn3
 
-comma:=,
 GCC_BUILD_DIR2:=$(TOOL_BUILD_DIR)/gcc-$(GCC_VERSION)-final
 $(GCC_BUILD_DIR2)/.configured: $(GCC_DIR)/.patched $(GCC_STAGING_PREREQ)
 	mkdir -p $(GCC_BUILD_DIR2)
--
cgit v0.8.2.1
